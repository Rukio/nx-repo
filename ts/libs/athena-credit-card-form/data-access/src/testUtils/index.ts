export * from './setupTestStore';
export * from './helpers';
