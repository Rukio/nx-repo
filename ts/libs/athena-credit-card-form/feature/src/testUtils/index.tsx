import { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { configureStore, ConfigureStoreOptions } from '@reduxjs/toolkit';
import { authSlice } from '@*company-data-covered*/auth0/data-access';
import {
  athenaCreditCardFormApiSlice,
  collectPaymentSlice,
  creditCardSlice,
  RootState,
} from '@*company-data-covered*/athena-credit-card-form/data-access';
import { setupListeners } from '@reduxjs/toolkit/query';
import { Provider } from 'react-redux';
import type { Config as UserEventSetupConfig } from '@testing-library/user-event/setup/setup';
import { MemoryRouter, MemoryRouterProps } from 'react-router-dom';

type CustomRenderOptions = {
  renderOptions?: Omit<RenderOptions, 'queries'>;
  userEventOptions?: Omit<UserEventSetupConfig, 'delay'>;
  withRouter?: boolean;
  routerProps?: MemoryRouterProps;
  preloadedState?: ConfigureStoreOptions<RootState>['preloadedState'];
};

const customRender = (
  ui: ReactElement,
  options: CustomRenderOptions = {
    withRouter: true,
  }
) => {
  const {
    withRouter,
    routerProps = {},
    userEventOptions,
    renderOptions,
    preloadedState,
  } = options;

  const store = configureStore({
    reducer: {
      [authSlice.name]: authSlice.reducer,
      [athenaCreditCardFormApiSlice.reducerPath]:
        athenaCreditCardFormApiSlice.reducer,
      [collectPaymentSlice.name]: collectPaymentSlice.reducer,
      [creditCardSlice.name]: creditCardSlice.reducer,
    },
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        immutableCheck: false,
        serializableCheck: false,
      }).concat(athenaCreditCardFormApiSlice.middleware),
    preloadedState,
  });

  setupListeners(store.dispatch);

  const component = withRouter ? (
    <MemoryRouter {...routerProps}>{ui}</MemoryRouter>
  ) : (
    ui
  );

  return {
    user: userEvent.setup({ ...userEventOptions, delay: null }),
    ...render(<Provider store={store}>{component}</Provider>, renderOptions),
  };
};

export * from '@testing-library/react';
export { customRender as render, userEvent };
