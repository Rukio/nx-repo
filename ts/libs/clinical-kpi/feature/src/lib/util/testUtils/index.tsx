import { DefaultBodyType, DelayMode, rest } from 'msw';
import {
  render as rtlRender,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import {
  RootState,
  store as clinicalKpiStore,
  AUTH_FEATURE_KEY,
  authSlice,
  CLINICAL_KPI_API_SLICE_KEY,
  clinicalKpiApiSlice,
  PEER_RANKINGS_KEY,
  peerRankingsSlice,
  PROVIDER_VISITS_TABLE_KEY,
  providerVisitsTableSlice,
  CARE_TEAM_RANKINGS_KEY,
  careTeamRankingsSlice,
  PROVIDER_SHIFTS_KEY,
  providerShiftsTableSlice,
  INDIVIDUAL_PERFORMANCE_POSITION_KEY,
  individualPerformancePositionSlice,
} from '@*company-data-covered*/clinical-kpi/data-access';
import UserProvider from '../../UserProvider';
import { server } from './server';

type RenderOptions = RTLRenderOptions & {
  preloadedState?: Partial<RootState>;
  store?: typeof clinicalKpiStore;
};

export function render(
  ui: React.ReactElement,
  {
    preloadedState,
    store = configureStore({
      reducer: {
        [AUTH_FEATURE_KEY]: authSlice.reducer,
        [CLINICAL_KPI_API_SLICE_KEY]: clinicalKpiApiSlice.reducer,
        [PEER_RANKINGS_KEY]: peerRankingsSlice.reducer,
        [PROVIDER_VISITS_TABLE_KEY]: providerVisitsTableSlice.reducer,
        [CARE_TEAM_RANKINGS_KEY]: careTeamRankingsSlice.reducer,
        [PROVIDER_SHIFTS_KEY]: providerShiftsTableSlice.reducer,
        [INDIVIDUAL_PERFORMANCE_POSITION_KEY]:
          individualPerformancePositionSlice.reducer,
      },
      preloadedState,
      middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
          immutableCheck: false,
          serializableCheck: false,
        }).concat(clinicalKpiApiSlice.middleware),
    }),
    ...renderOptions
  }: RenderOptions = {}
) {
  setupListeners(store.dispatch);
  const Wrapper: React.FC<{ children: ReactNode }> = ({ children }) => {
    return (
      <BrowserRouter>
        <Provider store={store}>{children}</Provider>
      </BrowserRouter>
    );
  };

  return {
    user: userEvent.setup({ delay: null }),
    ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }),
    store,
  };
}

export const renderWithUserProvider = (
  component?: React.ReactNode,
  renderOptions?: RenderOptions
) => {
  return {
    ...render(<UserProvider>{component}</UserProvider>, renderOptions),
  };
};

type InterceptQueryProps = {
  url: string;
  data?: DefaultBodyType;
  delay?: DelayMode | number;
  statusCode?: number;
};
// TODO: [PT-1289] find a way to make mock intercept builders type-safe for each endpoint
export const interceptQuery = ({
  url,
  data,
  delay,
  statusCode,
}: InterceptQueryProps) => {
  server.use(
    rest.get(url, (_req, res, ctx) =>
      res.once(ctx.status(statusCode || 200), ctx.delay(delay), ctx.json(data))
    )
  );
};

export * from '@testing-library/react';
