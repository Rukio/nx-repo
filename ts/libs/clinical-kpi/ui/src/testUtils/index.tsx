import { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';

type CustomRenderOptions = {
  withRouter?: boolean;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  { withRouter = false }: CustomRenderOptions = {}
) => {
  const component = withRouter ? <BrowserRouter>{ui}</BrowserRouter> : ui;

  return {
    user: userEvent.setup(),
    ...render(component, options),
  };
};

export * from '@testing-library/react';
export { customRender as render, userEvent };
