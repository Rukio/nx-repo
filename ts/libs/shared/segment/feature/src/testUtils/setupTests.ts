import '@*company-data-covered*/shared/testing/react/setup';
