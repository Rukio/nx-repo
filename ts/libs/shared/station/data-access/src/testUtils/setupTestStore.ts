import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { authSlice, AUTH_FEATURE_KEY } from '@*company-data-covered*/auth0/data-access';
import { stationApiSlice } from '../';

export const setupTestStore = () =>
  configureStore({
    reducer: combineReducers({
      [AUTH_FEATURE_KEY]: authSlice.reducer,
      [stationApiSlice.reducerPath]: stationApiSlice.reducer,
    }),
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat(stationApiSlice.middleware),
  });
