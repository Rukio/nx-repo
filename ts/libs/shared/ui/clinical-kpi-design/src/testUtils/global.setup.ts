module.exports = async () => {
  process.env.TZ = process.env.TZ || 'America/Denver';
};
