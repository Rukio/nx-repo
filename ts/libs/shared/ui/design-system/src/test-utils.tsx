import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider, theme } from './lib';
import type { Config as UserEventSetupConfig } from '@testing-library/user-event/setup/setup';
import userEvent from '@testing-library/user-event';

type CustomRenderOptions = {
  renderOptions?: Omit<RenderOptions, 'wrapper'>;
  userEventOptions?: UserEventSetupConfig;
};

const WrappedProviders: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: React.ReactElement,
  options?: CustomRenderOptions
) => {
  return {
    user: userEvent.setup(options?.userEventOptions),
    ...render(ui, { wrapper: WrappedProviders, ...options?.renderOptions }),
  };
};

export * from '@testing-library/react';
export { customRender as render, userEvent };
