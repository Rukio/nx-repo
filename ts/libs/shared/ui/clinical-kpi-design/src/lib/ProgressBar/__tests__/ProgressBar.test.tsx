import { render } from '../../../testUtils/test-utils';
import {
  BasicProgressBar,
  NoTextProgressBar,
} from '../__storybook__/ProgressBar.stories';

describe('Snapshot tests', () => {
  test('should render BasicProgressBar', () => {
    const { asFragment } = render(<BasicProgressBar />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should render NoTextProgressBar', () => {
    const { asFragment } = render(<NoTextProgressBar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
