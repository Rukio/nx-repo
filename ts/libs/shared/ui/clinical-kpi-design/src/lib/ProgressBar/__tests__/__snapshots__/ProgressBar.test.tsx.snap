// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests should render BasicProgressBar 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  box-sizing: border-box;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2px;
}

.emotion-2 {
  width: 24px;
  height: 12px;
  border-radius: 4px;
  background-color: #4caf50;
}

.emotion-9 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.66;
  letter-spacing: 0.03333em;
  color: rgba(0, 0, 0, 0.87);
}

<div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiGrid-root emotion-1"
    >
      <div
        class="MuiBox-root emotion-2"
      />
      <div
        class="MuiBox-root emotion-2"
      />
      <div
        class="MuiBox-root emotion-2"
      />
      <div
        class="MuiBox-root emotion-2"
      />
      <div
        class="MuiBox-root emotion-2"
      />
      <div
        class="MuiBox-root emotion-2"
      />
      <div
        class="MuiBox-root emotion-2"
      />
    </div>
    <span
      class="MuiTypography-root MuiTypography-caption emotion-9"
    >
      You cared for 7 patients
    </span>
  </div>
</DocumentFragment>
`;

exports[`Snapshot tests should render NoTextProgressBar 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  box-sizing: border-box;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2px;
}

.emotion-2 {
  width: 24px;
  height: 12px;
  border-radius: 4px;
  background-color: #4caf50;
}

<div
    class="MuiBox-root emotion-0"
  >
    <div
      class="MuiGrid-root emotion-1"
    >
      <div
        class="MuiBox-root emotion-2"
      />
      <div
        class="MuiBox-root emotion-2"
      />
      <div
        class="MuiBox-root emotion-2"
      />
      <div
        class="MuiBox-root emotion-2"
      />
      <div
        class="MuiBox-root emotion-2"
      />
      <div
        class="MuiBox-root emotion-2"
      />
      <div
        class="MuiBox-root emotion-2"
      />
    </div>
  </div>
</DocumentFragment>
`;
